src/components/show-products/index.js

```diff
@@ -155,7 +155,8 @@ export const ShowProducts = () => {
                     id={id}
                     name={productName}
                     onChange={(event) => handleCheckboxChange(event, id)}
-                    checked={checked}
+                    checked={!!checked}
+                    disabled={checked}
                     aria-label={TimeFrameLabels[timeFrameLabel]}
```

Comment:
I like that you're using the `disabled` and `checked` properties. Now, it's clearer and simpler to understand. 💯 🎊